experimental/sdg-tracker/src/components/search/Search.tsx

```diff
@@ -245,22 +245,24 @@ const Search = () => {
     async (searchQuery?: string) => {
       setIsSearching(true);
       setErrorMessage("");
-      const responseObj = await dataCommonsClient.detect(searchQuery || query);
+      const fulfillResponse = await dataCommonsClient.detectAndFulfill(
+        searchQuery || query
+      );
       setIsSearching(false);
 
-      if (responseObj.failure) {
-        setErrorMessage(responseObj.failure);
+      if (fulfillResponse.failure || fulfillResponse.userMessage) {
```

Comment:
userMessage and failure should be handled differently? Because user message can be for fallback in which case I think we would still want to show charts? e.g., user message can be something like "we couldn't detect a place, so showing charts for world"

Reply:
done